test/e2e/lib/framework/createTest.ts

```diff
@@ -182,53 +189,97 @@ function declareTestsForSetups(
 
     const setup = factory(setupOptions, servers)
     servers.base.bindServerApp(createMockServerApp(servers, setup))
     servers.crossOrigin.bindServerApp(createMockServerApp(servers, setup))
 
-    await setUpTest(testContext)
+    await setUpTest(browserLogs, testContext)
 
     try {
       await runner(testContext)
+      tearDownPassedTest(testContext)
     } finally {
       await tearDownTest(testContext)
-      log(`End '${spec.getFullName()}'`)
     }
   })
 }
 
-function createTestContext(servers: Servers, { basePath }: SetupOptions): TestContext {
+function createTestContext(
+  servers: Servers,
+  page: Page,
+  browserContext: BrowserContext,
+  browserLogsManager: BrowserLogsManager,
+  browserName: TestContext['browserName'],
+  { basePath }: SetupOptions
+): TestContext {
   return {
     baseUrl: servers.base.url + basePath,
     crossOriginUrl: servers.crossOrigin.url,
     intakeRegistry: new IntakeRegistry(),
     servers,
+    page,
+    browserContext,
+    browserName,
+    withBrowserLogs: (cb: (logs: BrowserLog[]) => void) => {
+      try {
+        cb(browserLogsManager.get())
+      } finally {
+        browserLogsManager.clear()
+      }
+    },
+    flushBrowserLogs: () => {
+      browserLogsManager.clear()
+    },
+    flushEvents: () => flushEvents(page),
+    deleteAllCookies: () => deleteAllCookies(browserContext),
+    sendXhr: (url: string, headers?: string[][]) => sendXhr(page, url, headers),
   }
 }
 
-async function setUpTest({ baseUrl }: TestContext) {
-  await browser.url(baseUrl)
+async function setUpTest(browserLogsManager: BrowserLogsManager, { baseUrl, page, browserContext }: TestContext) {
+  browserContext.on('console', (msg) => {
+    browserLogsManager.add({
+      level: msg.type() as BrowserLog['level'],
+      message: msg.text(),
+      source: 'console',
+      timestamp: Date.now(),
+    })
+  })
+
+  browserContext.on('weberror', (webError) => {
+    browserLogsManager.add({
+      level: 'error',
+      message: webError.error().message,
+      source: 'console',
+      timestamp: Date.now(),
+    })
+  })
+
+  await page.goto(baseUrl)
   await waitForServersIdle()
 }
 
-async function tearDownTest({ intakeRegistry }: TestContext) {
-  await flushEvents()
-  expect(intakeRegistry.telemetryErrorEvents).toEqual([])
+function tearDownPassedTest({ intakeRegistry, withBrowserLogs }: TestContext) {
+  expect(intakeRegistry.telemetryErrorEvents).toHaveLength(0)
   validateRumFormat(intakeRegistry.rumEvents)
-  await withBrowserLogs((logs) => {
-    logs.forEach((browserLog) => {
-      log(`Browser ${browserLog.source}: ${browserLog.level} ${browserLog.message}`)
-    })
-    expect(logs.filter((l) => (l as any).level === 'SEVERE')).toEqual([])
+  withBrowserLogs((logs) => {
+    expect(logs.filter((log) => log.level === 'error')).toHaveLength(0)
   })
+}
+
+async function tearDownTest({ flushEvents, deleteAllCookies }: TestContext) {
+  await flushEvents()
   await deleteAllCookies()
 }
```

Comment:
Split `tearDownTest` and `tearDownPassedTest` so that when a test fails no more `expect`  than the one failing is executed. This is because if the log shows only the last failing expect, hence if the teardown expect fails, it will show only this log which make debugging failing test difficult.